### ISSUE TITLE
Closure syntax allowed in pearlite!{} but not in proof_assert!{}

### ISSUE THREAD

**Issue by Armael** [creusot-rs/creusot#1215]
It look like the closure syntax is allowed in `pearlite!{}` blocks to define `Mapping`s, but when trying to use it in a `proof_assert!{}` I get an error from creusot.

```rust
#[logic]
#[ensures(true)]
fn foo() {
    proof_assert!{
        let f: Mapping<Int,Int> = { |k:Int| 42 };
        true
    };
}
```
raises the error `Closure(path/to/file.rs)`

**Comment by Armael** [comment]
Actually, `proof_assert!` looks like a red herring here. I'm getting the same issue with the following program:
```
#[logic]
#[ensures(true)]
fn foo() -> Seq<Int> { pearlite!{
    Seq::create(42, { |k:Int| 42 })
}}
```

**Comment by Armael** [comment]
Ah, the `#[ensures(true)]` seems to be required for the error to appear.

**Comment by xldenis** [comment]
Yes, the logic vcgenerator doesn't support closures.

**Comment by xldenis** [comment]
the VC it generates is a bit more complex than this, but that is the principle, the issue though is what happens if your closure contains a recursive call? what if it contains an expression with preconditions etc..

**Comment by jhjourdan** [comment]
YEah, but we should really do something (even if stupid) with every language construct, including closures.

**Comment by jhjourdan** [comment]
The VCgen now supports closures.
